src/composables/airGap.ts

```diff
@@ -0,0 +1,157 @@
+import { ref } from 'vue';
+import { Encoded } from '@aeternity/aepp-sdk';
+import { UR, UREncoder } from '@ngraveio/bc-ur';
+import bs58check from 'bs58check';
+import { AeternityModule } from '@airgap/aeternity';
+import {
+  MainProtocolSymbols,
+  IACMessageType,
+  AccountShareResponse,
+  AeternityProtocol,
+} from 'airgap-coin-lib';
+import { SerializerV3, IACMessageDefinitionObjectV3 } from '@airgap/serializer';
+
+import type { IAccountRaw } from '@/types';
+import { handleUnknownError, watchUntilTruthy } from '@/utils';
+import { ACCOUNT_AIR_GAP_WALLET, MOBILE_SCHEMA, PROTOCOLS } from '@/constants';
+
+const isComposableInitialized = ref(false);
+const serializer = ref<SerializerV3>();
+
+export function useAirGap() {
+  (async () => {
+    if (isComposableInitialized.value) {
+      return;
+    }
+    isComposableInitialized.value = true;
+
+    const serializerV3Companion = await new AeternityModule().createV3SerializerCompanion();
+    serializerV3Companion.schemas.forEach((schema) => {
+      SerializerV3.addSchema(schema.type, schema.schema, MainProtocolSymbols.AE);
+    });
+    serializer.value = SerializerV3.getInstance();
+  })();
+
+  /**
+   * Encodes an array of IACMessageDefinitionObjectV3 objects into a UR string.
+   */
+  async function encodeIACMessageDefinitionObjects(data: IACMessageDefinitionObjectV3[]) {
+    await watchUntilTruthy(serializer);
+    try {
+      const serializedData = await serializer.value?.serialize(data);
+
+      const dataUint8Array = bs58check.decode(serializedData!);
+      const ur = UR.fromBuffer(Buffer.from(dataUint8Array));
+
+      // Set the chunk sizes for single-chunk and multi-chunk encoding.
+      const SETTINGS_SERIALIZER_SINGLE_CHUNK_SIZE = 500;
+      const SETTINGS_SERIALIZER_MULTI_CHUNK_SIZE = 250;
+
+      // Create a UR encoder for single-chunk encoding.
+      const singleEncoder = new UREncoder(
+        ur,
+        SETTINGS_SERIALIZER_SINGLE_CHUNK_SIZE,
+      );
+      // If the UR requires multi-chunk encoding,
+      // create a new UR encoder with the appropriate chunk size.
+      if (singleEncoder.fragmentsLength !== 1) {
+        const multiEncoder = new UREncoder(
+          ur,
+          SETTINGS_SERIALIZER_MULTI_CHUNK_SIZE,
+        );
+        const fragments: string[] = [];
+
+        [...Array(multiEncoder.fragmentsLength)].forEach(() => {
+          fragments.push(multiEncoder.nextPart());
+        });
+
+        return fragments;
```

Comment:
```suggestion
        return [...Array(multiEncoder.fragmentsLength)].map(() => multiEncoder.nextPart());
```

Not tested, just a rough idea.